#### Problem Statement 1:
Blood glucose levels for obese patients have a mean of 100 with a standard deviation of 15. A researcher thinks that a diet high in raw cornstarch will have a positive effect on
blood glucose levels. A sample of 36 patients who have tried the raw cornstarch diet
have a mean glucose level of 108. Test the hypothesis that the raw cornstarch had an
effect or not.

##### H0: mu =100
##### H1: mu!=100

In [4]:
p_mean=100
p_std=12
n=36
s_mean=108

In [12]:
import numpy as np
from scipy import stats

In [6]:
def zscore_1(mu, sigma, x_bar, n):
    return (x_bar-mu)/(sigma/np.sqrt(n))

In [7]:
z_score=zscore_1(p_mean,p_std,s_mean,n)
z_score

4.0

In [8]:
alpha=0.05
p=1-alpha/2
critical=stats.norm.ppf(abs(p))
critical

1.959963984540054

In [9]:
if(abs(z_score)>critical):
    print("H0 Rejected")
else:
    print("H0 accepted")

H0 Rejected


#### Problem Statement 2:
In one state, 52% of the voters are Republicans, and 48% are Democrats. In a second
state, 47% of the voters are Republicans, and 53% are Democrats. Suppose a simple
random sample of 100 voters are surveyed from each state.
What is the probability that the survey will show a greater percentage of Republican
voters in the second state than in the first state?

##### H0 : p_1 >= p_2
##### H1 : p_1<p_2

In [10]:
def get_z_score(p_1,p_2,n1,n2,D=0): 
    #D difference in population proportion
    numerator = (p_1-p_2)-D
    denominator = np.sqrt(p_1*(1-p_1)/n1 +p_2*(1-p_2)/n2)
    z_value = numerator/denominator
    return z_value

In [11]:
p_1 = 0.52
p_2 = 0.47
n1 = 100
n2 = 100

In [13]:
zscore=get_z_score(p_1,p_2,n1,n2)

In [14]:
zscore

0.7080278164104213

In [16]:
p_value=1 - stats.norm.cdf(abs(zscore))
print("Probability of greater percentage of Republican voters in second state than in first state: {}".format(p_value))

Probability of greater percentage of Republican voters in second state than in first state: 0.23946399182220013


#### Problem Statement 3:
You take the SAT and score 1100. The mean score for the SAT is 1026 and the standard
deviation is 209. How well did you score on the test compared to the average test taker?

In [19]:
z_score = zscore_1(1026,209,1100,n=1)
z_score

0.35406698564593303

##### I scored 0.35 standard deviation times above than the average test taker